### PR TITLE
Boring to Beautiful codelab

### PR DESCRIPTION
**Pull request by khanhnwin**
TODO: Will update the readme with proper information and Unsplash image credits!

### CODE REVIEW

**Comment by domesticmouse** [comment]
Am I reading the logs correctly in guessing this should be marked up Dart min version as 2.17.0-0? 

Context: https://github.com/flutter/codelabs/runs/5708612711?check_suite_focus=true#step:7:41

**Comment by khanhnwin** [comment]
Now it looks like it's failing waiting on fix from #288?

**Comment by domesticmouse** [comment]
I'd prefer to get this working on stable, if possible. Otherwise this repo is going to be red until 2.17 goes stable.

File: boring_to_beautiful/final/pubspec.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ publish_to: "none"
 version: 1.0.0
 
 environment:
-  sdk: ">=2.17.0-0"
+  sdk: ^2.12.0
```
Comment (domesticmouse) [inline boring_to_beautiful/final/pubspec.yaml:7]: I'm happy for this to be ^2.16.0 or even ^2.16.1.

Comment (domesticmouse) [reply boring_to_beautiful/final/pubspec.yaml:7]: (I believe `flutter create` on stable is currently generatinng 2.16.1 as minimum version.)

Comment (khanhnwin) [reply boring_to_beautiful/final/pubspec.yaml]: Fixed!

**Comment by khanhnwin** [comment]
> I'd prefer to get this working on stable, if possible. Otherwise this repo is going to be red until 2.17 goes stable.

Set dart version to 2.16.1. The conflict remains for material_color_utilities version. So Flutter stable depends on 0.1.3, but dynamic_color depends on ^0.1.4 - this is resolved in beta. Any idea on how we can work around that?

**Comment by domesticmouse** [comment]
I'm in the process landing https://github.com/flutter/codelabs/pull/294 which will fix the beta channel issues.

**Comment by domesticmouse** [comment]
Also, please add this codelab to https://github.com/flutter/codelabs/blob/boring-to-beautiful/flutter_ci_script_beta.sh#L8

**Review by domesticmouse — COMMENTED** [review]
This codelab is beta only, because of dependencies.

File: boring_to_beautiful/final/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: myartist
+description: A new Flutter project.
+publish_to: "none"
+version: 1.0.0
+
+environment:
+  sdk: ^2.16.1
```
[the remark this answers is not in the document]
Comment (khanhnwin) [reply boring_to_beautiful/final/pubspec.yaml]: Done!

**Comment by khanhnwin** [comment]
- Added code lab to ci script: https://github.com/flutter/codelabs/pull/293/commits/9940fdeecd1f6c357355ca3f7fb91814c1958b3d
- Set dart min version to ^2.17.0-0
- update dynamic_color to ^1.2.0 which has the expected API

**Comment by domesticmouse** [comment]
I don't understand why this is failing the stable CI. I can't land this until stable goes green

**Comment by khanhnwin** [comment]
https://github.com/flutter/codelabs/blob/boring-to-beautiful/flutter_ci_script_shared.sh is going through all the projects and running `flutter pub get`!

**Review by domesticmouse — APPROVED** [review]

**Comment by khanhnwin** [comment]
Hey @amejiarosario thanks for the heads up on that! Checking it out now!

**Comment by khanhnwin** [comment]
Follow up: Just confirmed this is fixed with #1004. Please let me know if you're still having trouble with it!

**Pull request merged**